src/common_conditions.rs

```diff
@@ -0,0 +1,40 @@
+//! Run conditions for actions.
+
+use crate::{prelude::ActionState, Actionlike};
+use bevy::prelude::Res;
+
+/// Stateful run condition that can be toggled via an action press using [`ActionState::just_pressed`].
+pub fn action_toggle_active<T>(default: bool, action: T) -> impl FnMut(Res<ActionState<T>>) -> bool
+where
+    T: Actionlike + Copy,
```

Comment:
I'd like to relax these bounds to Clone: there's no reason to enforce Copy here.

Reply:
But we will clone it every usage because `pressed` and similar take by value. Is it okay?

Reply:
Not any worse than the `Copy` bound :) In either case, it's getting copied every frame.